plugins/VstEffect/CMakeLists.txt

```diff
@@ -1,8 +1,13 @@
 IF(LMMS_SUPPORT_VST)
 INCLUDE(BuildPlugin)
 INCLUDE_DIRECTORIES("${CMAKE_CURRENT_SOURCE_DIR}/../vst_base")
-LINK_DIRECTORIES("${CMAKE_CURRENT_BINARY_DIR}/../vst_base")
-LINK_LIBRARIES(vstbase)
+LINK_DIRECTORIES("${CMAKE_CURRENT_BINARY_DIR}/..")
+IF(LMMS_BUILD_WIN32)
+	LINK_LIBRARIES(vstbase)
+ELSE()
+	LINK_LIBRARIES(vstbase -Wl,--enable-new-dtags)
```

Comment:
Thinking out loud here...

Had to look this one up... Since [`enable-new-dtags` is an ELF-only feature](http://www.delorie.com/gnu/docs/binutils/ld_3.html) it won't work for Win32, hence the `IF(...)` 👍 

> From Free Software Foundation:
> This linker can create the new dynamic tags in ELF. But the older ELF systems may not understand them. If you specify `--enable-new-dtags', the dynamic tags will be created as needed. If you specify `--disable-new-dtags', no new dynamic tags will be created. By default, the new dynamic tags are not created. Note that those options are only available for ELF systems.

> From wikepedia:
> If the new-dtags feature is enabled in the linker (at run time using --enable-new-dtags), GNU ld, besides setting the DT_RPATH attribute, also sets the DT_RUNPATH attribute to the same string. At run time, if the dynamic linker finds a DT_RUNPATH attribute, it ignores the value of the DT_RPATH attribute, with the effect that LD_LIBRARY_PATH is checked next and the paths in the DT_RUNPATH attribute are only searched after it.

> This means that in such configurations, the paths in LD_LIBRARY_PATH are searched before those given at link time using -rpath if --enable-new-dtags was active.

> The ld dynamic linker does not search DT_RUNPATH locations for transitive dependencies, unlike DT_RPATH. [3]

> Instead of specifying the -rpath to the linker, the environment variable LD_RUN_PATH can be set to the same effect.

I'm still a bit confused by the flag as it seems it would honor a system library path for these shared object before the LMMS library path, but perhaps I'm misunderstanding.

Regardless, I'll test things out and report back. 👍   I can confirm that when I run LMMS in DependencyWalker and Wine, I can see the software searching for the library in the wrong location each time it starts up, so this is something we've needed for a while, thanks.